arviz/data/inference_data.py

```diff
@@ -324,6 +325,59 @@ def sel(
         else:
             return out
 
+    def add_groups(self, group_dict=None, coords=None, dims=None, **kwargs):
+        """Add new groups to InferenceData object.
+
+        Parameters
+        ----------
+        group_dict: dict[str] -> dict, xr.Dataset
+            Groups to be added
+        coords : dict[str] -> ndarray
+            Coordinates for the dataset
+        dims : dict[str] -> list[str]
+            Dimensions of each variable. The keys are variable names, values are lists of
+            coordinates.
+        **kwargs: mapping
+            The keyword arguments form of group_dict. One of group_dict or kwargs must be provided.
+
+        """
+        if group_dict is None:
+            group_dict = {}
+
+        group_dict.update(kwargs)
+        if not group_dict:
+            raise ValueError("One of group_dict or kwargs must be provided.")
+        for group, dataset in group_dict.items():
+            if group in self._groups_all:
+                warnings.warn(
+                    "{group} group already exists. Continuing without adding the group {group}".format(
+                        group=group
+                    ),
+                    UserWarning,
+                )
```

Comment:
isn't a `continue` needed here to actually skip the group? I am more inclined towards erroring though, more or less like concat does.

Reply:
Sorry, it was intended to add continue here. My mistake. Will raise error!